Userland/Libraries/LibGL/Tex/Texture.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Format.h>
+#include <LibGL/GL/gl.h>
+#include <LibGL/Tex/Texture.h>
+#include <string.h>
+
+namespace GL
+{
+
+void Texture::upload_texture_data([[maybe_unused]] GLenum target, GLint lod, GLint internal_format, GLsizei width, GLsizei height, [[maybe_unused]] GLint border, GLenum format, GLenum type, const GLvoid *pixels)
+{
+    // NOTE: Some target, format, and internal formats are currently unsupported.
+    // Considering we control this library, and `gl.h` itself, we don't need to add any
+    // checks here to see if we support them; the program will simply fail to compile..
+
+    if (lod > LOG2_MAX_TEXTURE_SIZE) {
+        dbgln("LibGL: Level of Detail exceeds maximum allowed");
+        return;
+    }
+
+    // Make sure user passed in support pixel data type
+    if (type != GL_UNSIGNED_BYTE) {
+        dbgln("LibGL: Unsupported pixel format type {}", type);
+        VERIFY_NOT_REACHED();
+    }
+
+    // Somebody passed us in nullptr...
+    if (pixels == nullptr) {
+        dbgln("LibGL: pixels == nullptr when uploading texture data.");
+        return;
```

Comment:
this seems like a great place to throw an `std::invalid_argument` exception but it seems that serenity os doesn't like using exceptions `:<`

Reply:
Exceptions are disabled, and will most likely remain disabled : )